### PR TITLE
openssh: update to 8.4p1, drop keychain patch and hpn variant

### PR DESCRIPTION
**Pull request by Schamschula**
update gsskex and macports-config patches

#### Description

openssh has been updated several times upstream since the last MacPorts update.
MacPorts has tried to maintain the unsupported Apple keychain patch.
Homebrew dropped support for this patch in 2016: https://archive.is/hSB6d#10%25

7.7p1, in 2018, was the last version to support the hpn option in FreeBSD Freshports:
https://www.freshports.org/security/openssh-portable/

###### Type(s)

- [x] update
- [x] security fix

###### Tested on

macOS 10.14.6 18G6042
Xcode 10.3 10G8

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by neverpanic** [comment]
Seems there's a bit of a mashup in the commit message now :)

**Comment by Schamschula** [comment]
Indeed! I should read my local wiki notes all the way though before pushing a squash commit...

I forgot to use --force!

Lots of unnecessary extra work.

**Comment by Schamschula** [comment]
Removed obsolete ```note``` content.

**Comment by neverpanic** [comment]
The commit message is still broken.

**Comment by Schamschula** [comment]
So it is!

I didn't catch that a premature press of the ```y``` key would continue to propagate.

File: net/openssh/Portfile [hunk 1]
```diff
@@ -50,9 +50,7 @@ if {${name} eq ${subport}} {
                         pam.patch \
                         patch-sandbox-darwin.c-apple-sandbox-named-external.diff \
                         patch-sshd.c-apple-sandbox-named-external.diff \
-                        0002-Apple-keychain-integration-other-changes.patch \
-                        macports-config.patch \
-                        patch-openbsd_compat-memmem-bug.diff
+                        macports-config.patch
 
     # We need a couple of patches
```
Comment (neverpanic) [inline net/openssh/Portfile:55]: This comment needs adjusting – it still references `0002-Apple-keychain-integration-other-changes.patch` that is now gone.

**Review by neverpanic — CHANGES_REQUESTED** [review]

**Review by neverpanic — APPROVED** [review]
lg2m now. Haven't tested all variants, though.

**Comment by Schamschula** [comment]
I did build all variants. However, I normally install the default variants.

**Pull request merged**